rslib/src/import_export/package/apkg/export.rs

```diff
@@ -0,0 +1,339 @@
+        self.extract_media_paths_from_notes(names, &notetypes)?;
+        self.extract_media_paths_from_notetypes(names, &notetypes);
+        Ok(())
+    }
+
+    fn extract_media_paths_from_notes(
+        &mut self,
+        names: &mut HashSet<PathBuf>,
+        notetypes: &HashMap<NotetypeId, Arc<Notetype>>,
+    ) -> Result<()> {
+        let mut stmt = self.storage.db.prepare("SELECT flds, mid FROM notes")?;
+        let mut rows = stmt.query([])?;
+        while let Some(row) = rows.next()? {
+            let flds = row.get_ref(0)?.as_str()?;
+            let notetype_id: NotetypeId = row.get(1)?;
+            self.extract_media_paths_from_note(names, flds, notetypes.get(&notetype_id).unwrap());
+        }
+        Ok(())
+    }
+
+    fn extract_media_paths_from_note(
+        &self,
+        names: &mut HashSet<PathBuf>,
+        flds: &str,
+        notetype: &Notetype,
+    ) {
+        self.extract_latex_paths(names, flds, notetype);
+        for media_ref in extract_media_refs(flds) {
+            if is_local_base_name(&media_ref.fname_decoded) {
+                names.insert(self.media_folder.join(media_ref.fname_decoded.as_ref()));
+            }
+        }
+    }
+
+    fn extract_latex_paths(&self, names: &mut HashSet<PathBuf>, flds: &str, notetype: &Notetype) {
+        for latex in extract_latex(flds, notetype.config.latex_svg).1 {
+            if is_local_base_name(&latex.fname) {
+                let path = self.media_folder.join(&latex.fname);
+                if path.exists() {
+                    names.insert(path);
+                }
+            }
+        }
+    }
+
+    fn extract_media_paths_from_notetypes(
+        &mut self,
+        names: &mut HashSet<PathBuf>,
+        notetypes: &HashMap<NotetypeId, Arc<Notetype>>,
+    ) {
+        for notetype in notetypes.values() {
+            notetype.extract_media_paths(names, &self.media_folder);
+        }
+    }
+}
+
+fn is_local_base_name(name: &str) -> bool {
+    !is_remote_filename(name) && Path::new(name).parent().is_none()
+}
+
+impl Notetype {
+    fn extract_media_paths(&self, names: &mut HashSet<PathBuf>, media_folder: &Path) {
+        for name in extract_underscored_css_imports(&self.config.css) {
+            if is_local_base_name(name) {
+                names.insert(media_folder.join(name));
+            }
+        }
+        for template in &self.templates {
+            template.extract_media_paths(names, media_folder);
+        }
+    }
+}
+
+impl CardTemplate {
+    fn extract_media_paths(&self, names: &mut HashSet<PathBuf>, media_folder: &Path) {
+        for template_side in [&self.config.q_format, &self.config.a_format] {
+            for name in extract_underscored_references(template_side) {
+                if is_local_base_name(name) {
+                    let path = media_folder.join(name);
+                    // shotgun approach, so check if paths actually exist
+                    if path.exists() {
```

Comment:
Not sure if it's worth it. If we only allow reasonable extensions, we'd need to differentiate between JS strings and HTML refs more strictly, and repeat the different quotation cases for both of them. And for every case, repeat the list of extensions.
If we just check for paths to exist, and ignore them if they don't, do we lose that much through false positives?

Anecdotally, I have a notetype with several kB of JS, generated by an add-on, and there's not a single match.

Reply:
I was more thinking of what would happen when we no longer require a leading underscore - we'd likely get a lot more matches then. Looking the files up when it's not a local folder is somewhat more expensive, so it would be good to avoid thousands of lookups that are likely to fail. But we can deal with that in the future when removing the underscores; for now it's probably fine.

Reply:
Actually, ignore me - we could relax the restriction but continue to require it for the bare reference case. We're in agreement then. :-)